java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -72,7 +72,7 @@ protected BaseAllocator(
     this(parentAllocator.listener, parentAllocator, name, initReservation, maxAllocation);
   }
 
-  private BaseAllocator(
+  protected BaseAllocator(
```

Comment:
Can we update the protected constructor/consolidate? Would rather not expose more constructors.

Reply:
I'm not sure how to update/consolidate, short of removing the previous two protected constructors -- which breaks the public interface, and thus probably not the best idea. I cannot use the existing constructors since I need to support listener instances that are not common to the entire hierarchy.
Suggestions are welcome.

Reply:
I think it is okay changing the constructor signature of BaseAllocator and consolidate overlap